src/aiai_eval/evaluator.py

```diff
@@ -244,12 +238,18 @@ def _evaluate_single(
             results = task(model_id)
             self.evaluation_results[task_config.name][model_id] = results
             logger.debug(f"Results:\n{results}")
-        except InvalidEvaluation as e:
-            logger.info(
-                f"{model_id} could not be evaluated on {task_config.pretty_name}. "
-                "Skipping."
-            )
-            logger.debug(f'The error message was "{e}".')
+        except (InvalidEvaluation, InvalidArchitectureForTask) as e:
+            if type(e) == InvalidEvaluation:
+                logger.info(
+                    f"{model_id} could not be evaluated on {task_config.pretty_name}. "
+                    "Skipping."
+                )
+                logger.debug(f'The error message was "{e}".')
+            elif type(e) == InvalidArchitectureForTask:
+                logger.warning(
+                    f"Skipping evaluation of {model_id} on {task_config.pretty_name} "
+                    "as the architecture is not supported by the task."
+                )
```

Comment:
Splitting these up in separate except blocks is a bit cleaner:

```suggestion
        except InvalidEvaluation:
            logger.info(
                f"{model_id} could not be evaluated on {task_config.pretty_name}. "
                "Skipping."
            )
            logger.debug(f'The error message was "{e}".')
        except InvalidArchitectureForTask:
            logger.warning(
                f"Skipping evaluation of {model_id} on {task_config.pretty_name} "
                "as the architecture is not supported by the task."
            )
```

Reply:
Also, why is the second exception handling a warning, with the first one being an info?